packages/data-stores/src/user/controls.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Internal dependencies
+ */
+import { wpcomRequest } from '../utils';
+import { CreateAccountAction } from './types';
+
+// @TODO: pass in locale, and remove magic strings for client_id and client_secret
+export default {
+	CREATE_ACCOUNT: async ( action: CreateAccountAction ) => {
```

Comment:
**TIL** about controls! Thank you! 

What would be the recommendation to "chain" HTTP requests?

For example, the mega rough plan is, once the user has handed over their email address, to:

- provision an account (using the work in this PR)
- create a site straight away

⚠️ n00b question

Is that something (an action?) we could dispatch from within a controls option object? E.g.,

```
PROVISION_USER_ACCOUNT_AND_SITE: async () => {
    ....
    const newUser = await userRequest();
    newUser?.id && dispatchSomeAction...
   ...
```

Reply:
I think that would probably work, but I was also wondering if it'd make sense for the site to live in its own store called something like `currentSite`? Then we'd be able to chain them together in Gutenboarding instead of in controls. I was imagining it could be something like:

In Gutenboarding:
* A user selects a design and page template(s), then clicks on "create site"
* A function then checks if there's a current user, and if so fires off `createSite`.
* If not, display the account creation form, and when it's submitted, fire off `createAccount`, and on success, fire off `createSite`. To check for success we could use a similar selector as to the hacky ["New user created!"](https://github.com/automattic/wp-calypso/blob/2b6e1888311f7b4659c806c16e35d20eb89b643b/client/landing/gutenboarding/components/header/index.tsx#L140) I've added in this PR.
* `createSite` creates the site, and on success stores the current site in the data store (so that we can deal with the duplicate sites issue further down the track), then redirect the user to the editor.

Reply:
> What would be the recommendation to "chain" HTTP requests?

On the technical level, this should be supported by `@wordpress/redux-routine` which is what enables the following patterns you'll see in all of our stores that are making API requests:

https://github.com/Automattic/wp-calypso/blob/0ac61f225a4e49d310f9348c7b02c2e36a60b320/packages/data-stores/src/verticals-templates/resolvers.ts#L11-L20

I believe you _could_ chain things right there:

```ts
function* doTheThings() {
  const { userId } = yield postUserCreateRequest();
  const { siteId } = yield postSiteRequest( { ownerId: userId } );
  // …
}
```

I'm not recommending it be done this way, just trying to share some familiarity with the data layer we're using.

---

> I was also wondering if it'd make sense for the site to live in its own store called something like `currentSite`?
> Then we'd be able to chain them together in Gutenboarding instead of in controls. I was imagining it could be something like […]

We've been wondering about when we should split up stores, and what the costs and benefits are. Nothing to report yet 🙂 

The benefit of breaking up should be making it easier to share a store for use without bloat, I'm not sure what the costs will be 🤔 More cumbersome interface?

The best we have to go on now is that store divisions should be made where they make sense, I think we should do our best to limit each store to s single concern.

What you outline seems reasonable. User and site creation actions are likely going to be useful in different contexts 👍